src/components/scenes/CoinRankingScene.tsx

```diff
@@ -0,0 +1,290 @@
+import { EdgeText } from '../themed/EdgeText'
+import { OutlinedTextInput, OutlinedTextInputRef } from '../themed/OutlinedTextInput'
+
+const coinRowData: CoinRowData = { coinRankings: [] }
+
+const QUERY_PAGE_SIZE = 30
+const LISTINGS_REFRESH_INTERVAL = 30000
+
+// Masking enable bit with 0 disables logging
+enableDebugLogType(LOG_COINRANK & 0)
+
+interface Props {
+  // navigation: NavigationProp<'coinRanking'>
+  // route: RouteProp<'coinRanking'>
+}
+
+const percentChangeOrder: PercentChangeTimeFrame[] = ['hours1', 'hours24', 'days7', 'days30', 'year1']
+const percentChangeStrings: { [pc: string]: string } = {
+  hours1: '1hr',
+  hours24: '24hr',
+  days7: '7d',
+  days30: '30d',
+  year1: '1y'
+}
+const priceSubTextStrings: { [pc: string]: string } = {
+  marketCap: 'MCap',
+  volume24h: 'Vol 24h'
+}
+
+type Timeout = ReturnType<typeof setTimeout>
+const CoinRankingComponent = (props: Props) => {
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const mounted = React.useRef<boolean>(true)
+  const textInput = React.useRef<OutlinedTextInputRef>(null)
+  const timeoutHandler = React.useRef<Timeout | undefined>()
+
+  const [requestDataSize, setRequestDataSize] = useState<number>(QUERY_PAGE_SIZE)
+  const [dataSize, setDataSize] = useState<number>(0)
+  const [searchText, setSearchText] = useState<string>('')
+  const [searching, setSearching] = useState<boolean>(false)
+  const [percentChangeTimeFrame, setPercentChangeTimeFrame] = useState<PercentChangeTimeFrame>('hours24')
+  const [priceSubText, setPriceSubText] = useState<PriceSubText>('marketCap')
+  const [fiatCode] = useState<string>('iso:USD')
+
+  const { coinRankings } = coinRowData
+
+  const renderItem = useHandler((itemObj: FlatListItem<number>) => {
+    const { index, item } = itemObj
+    const currencyCode = coinRankings[index]?.currencyCode ?? 'NO_CURRENCY_CODE'
+    const rank = coinRankings[index]?.rank ?? 'NO_RANK'
+    const key = `${index}-${item}-${rank}-${currencyCode}`
+    debugLog(LOG_COINRANK, `renderItem ${key.toString()}`)
+
+    return <CoinRankRow index={item} key={key} coinRowData={coinRowData} percentChangeTimeFrame={percentChangeTimeFrame} priceSubText={priceSubText} />
+  })
+
+  const handleEndReached = useHandler(() => {
+    debugLog(LOG_COINRANK, `handleEndReached. setRequestDataSize ${requestDataSize + QUERY_PAGE_SIZE}`)
+    setRequestDataSize(requestDataSize + QUERY_PAGE_SIZE)
+  })
+
+  const handlePercentChange = useHandler(() => {
+    const index = percentChangeOrder.indexOf(percentChangeTimeFrame)
+    if (index < 0) {
+      const msg = `Invalid percent change value ${percentChangeTimeFrame}`
+      console.error(msg)
+      showError(msg)
+      return
+    }
+    const newIndex = index + 1 >= percentChangeOrder.length ? 0 : index + 1
+    const newTimeFrame = percentChangeOrder[newIndex]
+    setPercentChangeTimeFrame(newTimeFrame)
+  })
+
+  const handlePriceSubText = useHandler(() => {
```

Comment:
Similar to the comment on handlePercentChange but more confusing. Vol and mcap are not similar categories yet they occupy the same  column. 

Since volume is given only in 24h terms, we potentially can replace the 2nd line of the 'Asset' space with the volume. I think it's acceptable to drop the name because it's common across all many crypto apps to hide names in an overview list. This proposal also aligns with Coinbase's 'trade' view.

Alternatively(1), we add a price column where we have subtext there that shows the volume, but this I'd prefer seeing percent change along with price instead.

Reply:
In order of importance IMO:
1. Currency Code OR Currency Name, not both
2. Price
3. % Change
4. 24h Vol
5. Mcap

More details like the name of the asset, market cap, can be hidden away until tapped

Reply:
If we go with a 2-row tile per item design, we can't really utilize two rows without the content contradicting the column title labels. The only exception to this is if we do a 2 row asset cell with name/CC, which I'm not really a fan of.
To make a 2-row design work, we'd need to move the title labels to the elements themselves (i.e. 'BTC\nVol 31.26k'), but then sorting would need to be condensed into one monolithic control. IMO this isn't very intuitive.

An argument could be made that if this is to be an overview, we can trim down the info and fit more items on the list by utilizing a 1 row design.